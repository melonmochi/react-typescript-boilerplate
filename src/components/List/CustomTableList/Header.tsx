import React, { FC, Dispatch } from "react";
import { Form, Switch, Radio } from "antd";
import { RadioChangeEvent } from "antd/lib/radio";
import { TableListState, showHeader, expandedRowRender } from ".";
import { useTranslation } from "react-i18next";

const scroll = { y: 240 };

interface HeaderProps {
  state: TableListState;
  setState: Dispatch<React.SetStateAction<TableListState>>;
}

export const Header: FC<HeaderProps> = props => {
  const { state, setState } = props;
  const { t } = useTranslation();

  const title = () => t("mock.title");
  const footer = () => t("mock.footer");

  const handleToggle = (prop: any) => (enable: boolean) => {
    setState({ ...state, [prop]: enable });
  };

  const handleSizeChange = (e: RadioChangeEvent) => {
    setState({ ...state, size: e.target.value });
  };

  const handleTableLayoutChange = (e: RadioChangeEvent) => {
    setState({ ...state, tableLayout: e.target.value });
  };

  const handleExpandChange = (enable: boolean) => {
    setState({
      ...state,
      expandedRowRender: enable ? expandedRowRender : undefined
    });
  };

  const handleEllipsisChange = (enable: boolean) => {
    setState({ ...state, ellipsis: enable });
  };

  const handleTitleChange = (enable: boolean) => {
    setState({ ...state, title: enable ? title : undefined });
  };

  const handleHeaderChange = (enable: boolean) => {
    setState({ ...state, showHeader: enable ? showHeader : false });
  };

  const handleFooterChange = (enable: boolean) => {
    setState({ ...state, footer: enable ? footer : undefined });
  };

  const handleRowSelectionChange = (enable: boolean) => {
    setState({ ...state, rowSelection: enable ? {} : undefined });
  };

  const handleScollChange = (enable: boolean) => {
    setState({ ...state, scroll: enable ? scroll : undefined });
  };

  const handleDataChange = (hasData: boolean) => {
    setState({ ...state, hasData });
  };

  const handlePaginationChange = (e: RadioChangeEvent) => {
    const { value } = e.target;
    setState({
      ...state,
      pagination: value === "none" ? false : { position: value }
    });
  };

  return (
    <Form
      layout="inline"
      className="components-table-demo-control-bar"
      style={{ marginBottom: 16 }}
    >
      <Form.Item label={t("tableConfig.bordered")}>
        <Switch checked={state.bordered} onChange={handleToggle("bordered")} />
      </Form.Item>
      <Form.Item label={t("tableConfig.loading")}>
        <Switch checked={state.loading} onChange={handleToggle("loading")} />
      </Form.Item>
      <Form.Item label={t("tableConfig.title")}>
        <Switch checked={!!state.title} onChange={handleTitleChange} />
      </Form.Item>
      <Form.Item label={t("tableConfig.columnHeader")}>
        <Switch checked={!!state.showHeader} onChange={handleHeaderChange} />
      </Form.Item>
      <Form.Item label={t("tableConfig.footer")}>
        <Switch checked={!!state.footer} onChange={handleFooterChange} />
      </Form.Item>
      <Form.Item label={t("tableConfig.expandable")}>
        <Switch
          checked={!!state.expandedRowRender}
          onChange={handleExpandChange}
        />
      </Form.Item>
      <Form.Item label={t("tableConfig.checkbox")}>
        <Switch
          checked={!!state.rowSelection}
          onChange={handleRowSelectionChange}
        />
      </Form.Item>
      <Form.Item label={t("tableConfig.fixedHeader")}>
        <Switch checked={!!state.scroll} onChange={handleScollChange} />
      </Form.Item>
      <Form.Item label={t("tableConfig.hasData")}>
        <Switch checked={!!state.hasData} onChange={handleDataChange} />
      </Form.Item>
      <Form.Item label={t("tableConfig.ellipsis")}>
        <Switch checked={!!state.ellipsis} onChange={handleEllipsisChange} />
      </Form.Item>
      <Form.Item label={t("tableConfig.size.title")}>
        <Radio.Group value={state.size} onChange={handleSizeChange}>
          <Radio.Button value="default">
            {t("tableConfig.size.default")}
          </Radio.Button>
          <Radio.Button value="middle">
            {t("tableConfig.size.middle")}
          </Radio.Button>
          <Radio.Button value="small">
            {t("tableConfig.size.small")}
          </Radio.Button>
        </Radio.Group>
      </Form.Item>
      <Form.Item label={t("tableConfig.tableLayout.title")}>
        <Radio.Group
          value={state.tableLayout}
          onChange={handleTableLayoutChange}
        >
          <Radio.Button value={undefined}>
            {t("tableConfig.tableLayout.unset")}
          </Radio.Button>
          <Radio.Button value="fixed">
            {t("tableConfig.tableLayout.fixed")}
          </Radio.Button>
        </Radio.Group>
      </Form.Item>
      <Form.Item label={t("tableConfig.pagination.title")}>
        <Radio.Group
          value={state.pagination ? state.pagination.position : "none"}
          onChange={handlePaginationChange}
        >
          <Radio.Button value="top">
            {t("tableConfig.pagination.top")}
          </Radio.Button>
          <Radio.Button value="bottom">
            {t("tableConfig.pagination.bottom")}
          </Radio.Button>
          <Radio.Button value="both">
            {t("tableConfig.pagination.both")}
          </Radio.Button>
          <Radio.Button value="none">
            {t("tableConfig.pagination.none")}
          </Radio.Button>
        </Radio.Group>
      </Form.Item>
    </Form>
  );
};
