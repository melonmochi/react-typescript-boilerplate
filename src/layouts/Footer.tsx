import React from "react";
import { Icon } from "antd";
import { DefaultFooter } from "@ant-design/pro-layout";

const Footer: React.FC = () => (
  <>
    <DefaultFooter
      copyright="2019 melonmochi"
      links={[
        {
          key: "React TypeScript Boilerplate",
          title: "React TypeScript Boilerplate",
          href: "https://github.com/melonmochi/react-typescript-boilerplate",
          blankTarget: true
        },
        {
          key: "github",
          title: <Icon type="github" />,
          href: "https://github.com/melonmochi/react-typescript-boilerplate",
          blankTarget: true
        },
        {
          key: "Paradigma Digital",
          title: "Paradigma Digital",
          href: "https://www.paradigmadigital.com/",
          blankTarget: true
        }
      ]}
    />
    <div
      style={{
        padding: "0px 24px 24px",
        textAlign: "center"
      }}
    >
      <a
        href="https://www.netlify.com"
        target="_blank"
        rel="noopener noreferrer"
      >
        <img
          src="https://www.netlify.com/img/global/badges/netlify-color-bg.svg"
          width="82px"
          alt="netlify logo"
        />
      </a>
    </div>
  </>
);

export default Footer;
