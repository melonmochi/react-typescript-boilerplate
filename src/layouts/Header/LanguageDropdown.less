@import "~antd/es/style/themes/default.less";

.dropdown {
  line-height: @layout-header-height;
  vertical-align: top;
  cursor: pointer;
  > i {
    font-size: 16px !important;
    transform: none !important;
    svg {
      position: relative;
      top: -1px;
    }
  }
}
