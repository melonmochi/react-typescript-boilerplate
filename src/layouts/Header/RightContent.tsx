import React from "react";
import { Icon, Tooltip } from "antd";
import { Search } from "@/components/Input";
import { LanguageDropdown } from ".";
import { ConnectProps } from "typings";
import styles from "./RightContent.less";

export type SiderTheme = "light" | "dark";
export type SiderLayout = "sidemenu" | "topmenu";
export interface GlobalHeaderRightProps extends ConnectProps {
  theme?: SiderTheme;
  layout?: SiderLayout;
}

const RightContent: React.SFC<GlobalHeaderRightProps> = props => {
  const { theme, layout } = props;
  let className = styles.right;

  if (theme === "dark" && layout === "topmenu") {
    className = `${styles.right}  ${styles.dark}`;
  }

  return (
    <div className={className}>
      <Search
        className={`${styles.action} ${styles.search}`}
        defaultActiveFirstOption={false}
        defaultOpen={false}
        // tslint:disable-next-line:jsx-no-lambda
        onChange={value => {
          console.log("change", value);
        }}
        // tslint:disable-next-line:jsx-no-lambda
        onSearch={value => {
          console.log("input", value);
        }}
        // tslint:disable-next-line:jsx-no-lambda
        onPressEnter={value => {
          console.log("enter", value);
        }}
        placeholder="Search"
      />
      <Tooltip title="Help">
        <a
          target="_blank"
          href="https://github.com/melonmochi/react-typescript-boilerplate"
          className={styles.action}
        >
          <Icon type="question-circle-o" />
        </a>
      </Tooltip>
      <LanguageDropdown className={styles.action} />
    </div>
  );
};

export default RightContent;
