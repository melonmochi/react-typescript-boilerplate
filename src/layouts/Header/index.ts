export { default as Dropdown } from "./Dropdown";
export { default as LanguageDropdown } from "./LanguageDropdown";
export { default as RightContent } from "./RightContent";
