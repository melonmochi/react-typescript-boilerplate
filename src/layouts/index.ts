export { default as BasicLayout } from "./BasicLayout";
export { default as Footer } from "./Footer";
