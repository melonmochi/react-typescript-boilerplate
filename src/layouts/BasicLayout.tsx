import React, { FC, useContext } from "react";
import ProLayout, {
  SettingDrawerProps,
  BasicLayoutProps as ProLayoutProps
} from "@ant-design/pro-layout";
import { BasicLayoutContext, BasicLayoutContextProvider } from "@/contexts";
import logo from "../assets/logo.png";
import Footer from "./Footer";
import { Link } from "react-router-dom";
import { RightContent } from "./Header";

export interface BasicLayoutProps extends ProLayoutProps {
  settings?: SettingDrawerProps["settings"];
}

export const BasicLayout: FC<BasicLayoutProps> = props => {
  const { state, dispatch } = useContext(BasicLayoutContext);
  const { collapsed } = state;
  const { settings } = props;

  const footerRender: BasicLayoutProps["footerRender"] = () => <Footer />;

  const handleMenuCollapse = (payload: boolean): void => {
    dispatch({ type: "CHANGE_COLLAPSED", payload });
  };

  const menuHeaderRender: BasicLayoutProps["menuHeaderRender"] = (
    logoDom,
    titleDom
  ) => (
    <Link to="/">
      {logoDom}
      {titleDom}
    </Link>
  );

  const rightContentRender: BasicLayoutProps["rightContentRender"] = rightProps => (
    <RightContent {...rightProps} />
  );

  return (
    <>
      <ProLayout
        collapsed={collapsed}
        footerRender={footerRender}
        logo={logo}
        menuHeaderRender={menuHeaderRender}
        onCollapse={handleMenuCollapse}
        rightContentRender={rightContentRender}
        title="React TypeScript"
        {...settings}
      />
    </>
  );
};

export const ConnectedBasicLayout: FC = () => {
  return (
    <BasicLayoutContextProvider>
      <BasicLayout />
    </BasicLayoutContextProvider>
  );
};

export default ConnectedBasicLayout;
