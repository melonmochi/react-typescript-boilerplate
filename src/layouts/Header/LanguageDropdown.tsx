import React from "react";
import classNames from "classnames";
import { Icon } from "antd";
import { Language } from "@/components/Selector";
import Dropdown from "./Dropdown";
import "./LanguageDropdown.less";

interface LanguageDropdownProps {
  className?: string;
}
const LanguageDropdown: React.FC<LanguageDropdownProps> = props => {
  const { className } = props;
  return (
    <Dropdown overlay={Language} placement="bottomRight">
      <span className={classNames("dropdown", className)}>
        <Icon type="global" title={"Languages"} />
      </span>
    </Dropdown>
  );
};

export default LanguageDropdown;
