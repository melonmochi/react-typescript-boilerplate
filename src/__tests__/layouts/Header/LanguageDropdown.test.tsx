import * as React from "react";
import { shallow } from "enzyme";
import { LanguageDropdown } from "@/layouts/Header";

describe("LanguageDropdown", () => {
  const container = shallow(<LanguageDropdown />);
  test("Snapshot", () => {
    expect(container).toMatchSnapshot();
  });
});
