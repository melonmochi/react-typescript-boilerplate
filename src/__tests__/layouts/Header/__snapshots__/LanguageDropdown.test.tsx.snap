// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageDropdown Snapshot 1`] = `
<HeaderDropdown
  overlay={[Function]}
  placement="bottomRight"
>
  <span
    className=""
  >
    <Icon
      title="Languages"
      type="global"
    />
  </span>
</HeaderDropdown>
`;
